### ISSUE TITLE
Loss is automatically converted?

### ISSUE THREAD

**Issue by ben-arnao** [tensorflow/probability#825]
From the bayesian_neural_network.py example-

```
# Model compilation.
  optimizer = tf.keras.optimizers.Adam(lr=FLAGS.learning_rate)
  # We use the categorical_crossentropy loss since the MNIST dataset contains
  # ten labels. The Keras API will then automatically add the
  # Kullback-Leibler divergence (contained on the individual layers of
  # the model), to the cross entropy loss, effectively
  # calcuating the (negated) Evidence Lower Bound Loss (ELBO)
  model.compile(optimizer, loss='categorical_crossentropy',
                metrics=['accuracy'], experimental_run_tf_function=False)
  return model
```

How does TF know to automatically convert the loss to probability and is converting a loss to probability compatible just a matter of adding KL to whatever loss you want to use?
